src/treetensornetworks/opsum_to_ttn.jl

```diff
@@ -38,6 +30,41 @@ function crosses_vertex(t::Scaled{C,Prod{Op}}, g::AbstractGraph, v) where {C}
   return v ∈ span(t, g)
 end
 
+function align_edges(edges, reference_edges)
+  return intersect(reference_edges, Iterators.flatten((edges, reverse.(edges))))
+end
+
+# return a dict from vertices `w` of `g`, except for `v`, to the incident edge of `v`
+# which lies in edge_path(g,w,v)
+function vertices_to_incident_edges_dict(g::AbstractGraph, v, incident_edges)
```

Comment:
It feels like this function is trying to do too many things in a single function, it is difficult for me to wrap my head around what it is doing and what it's purpose is.

Reply:
The description also feels underdefined. Can you try to refactor the function into smaller parts, and also rephrase what it is doing? Is the main idea that there is a path of vertices, and you want to output a dictionary from the vertices in that path to the incident edges of those vertices?

There is a function `boundary_edges` in `NamedGraphs.jl`: https://github.com/mtfishman/NamedGraphs.jl/blob/v0.1.25/src/Graphs/boundary.jl, does that relate to what this function is doing and can it help to simplify it? Perhaps it is some combination of `boundary_edges`, and then a `group` operation from https://github.com/JuliaData/SplitApplyCombine.jl which then groups those boundary edges by which ones share a vertex within the region? If it can be reframed in that way, ideally we wouldn't have it as a separate function, and instead just use a combination of `boundary_edges` and `group` inside `ttn_svd`. I think if a function is too specialized and not built around a simple and general conceptual framework that you can imagine using elsewhere, it isn't necessarily good to turn it into a separate function.

Reply:
I wasn't aware of `boundary_edges`. I can probably simplify the code using that and move it into `ttn_svd`.

Reply:
I simplified the code, and removed `span` in favor of reusing the `Dictionary` `which_incident_edges` to determine whether a term crosses a vertex. Unfortunately, it makes the code a little slower (but not by much), so once we have a faster algorithm for `span` we may want to switch the implementation back to the previous pattern of precomputing the `span` for every term.